app/src/main/java/es/wolfi/app/passman/CustomFieldEditAdapter.java

```diff
@@ -0,0 +1,282 @@
+            }
+
+            private Timer timer = new Timer();
+            private final long DELAY = 100; // milliseconds
+
+            @Override
+            public void afterTextChanged(Editable s) {
+                timer.cancel();
+                timer = new Timer();
+                timer.schedule(
+                        new TimerTask() {
+                            @Override
+                            public void run() {
+                                // you will probably need to use runOnUiThread(Runnable action) for some specific actions (e.g. manipulating views)
+                                int itemIndex = mValues.indexOf(holder.mItem);
+                                holder.mItem.setLabel(s.toString());
+                                mValues.set(itemIndex, holder.mItem);
+                            }
+                        },
+                        DELAY
+                );
+            }
+        });
+
+        holder.mValueEdit.addTextChangedListener(new TextWatcher() {
+
+            @Override
+            public void onTextChanged(CharSequence s, int start, int before, int count) {
+            }
+
+            @Override
+            public void beforeTextChanged(CharSequence s, int start, int count,
+                                          int after) {
+            }
+
+            private Timer timer = new Timer();
+            private final long DELAY = 100; // milliseconds
+
+            @Override
+            public void afterTextChanged(Editable s) {
+                timer.cancel();
+                timer = new Timer();
+                timer.schedule(
+                        new TimerTask() {
+                            @Override
+                            public void run() {
+                                // you will probably need to use runOnUiThread(Runnable action) for some specific actions (e.g. manipulating views)
+                                int itemIndex = mValues.indexOf(holder.mItem);
+                                holder.mItem.setValue(s.toString());
+                                mValues.set(itemIndex, holder.mItem);
+                            }
+                        },
+                        DELAY
+                );
+            }
+        });
+
+        holder.deleteButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Context context = holder.mView.getContext();
+
+                if (holder.mItem.getFieldType().equals("file")) {
+                    final ProgressDialog progress = new ProgressDialog(context);
+                    progress.setTitle(context.getString(R.string.loading));
+                    progress.setMessage(context.getString(R.string.wait_while_loading));
+                    progress.setCancelable(false);
+                    progress.show();
+
+                    AsyncHttpResponseHandler responseHandler = new AsyncHttpResponseHandler() {
```

Comment:
Could it make sense to move the API interaction to the API library on this context?

Reply:
replaced with one of the new response handlers